File: programming 1/lab 5/src/commands/Promptable.java
package commands;

import java.io.BufferedReader;

/**
 * Интерфейс, представляющий команды, требующие ввода данных элемента.
 */
public interface Promptable {
    /**
     * Выполняет команду, используя данные из файла.
     * @param reader для чтения данных из файла.
     * @param arg аргумент команды (может быть null).
     */
    void execute(BufferedReader reader, String arg);
}

File: programming 1/lab 6 kotlin/kotlin/lab 6/src/main/java/server/commands/Promptable.kt
package server.commands

import common.communication.Response
import java.io.BufferedReader
import java.io.IOException

/**
 * Интерфейс, представляющий команды, требующие ввода данных элемента.
 */
interface Promptable {
    /**
     * Выполняет команду, используя данные из файла.
     * @param reader для чтения данных из файла.
     * @param arg аргумент команды (может быть null).
     */
    fun execute(reader: BufferedReader, arg: String?): Response
}
